codec-http2/src/main/java/io/netty/handler/codec/http2/internal/hpack/Encoder.java

```diff
@@ -142,13 +142,13 @@ public void setMaxHeaderTableSize(ByteBuf out, int maxHeaderTableSize) {
         }
         capacity = maxHeaderTableSize;
         ensureCapacity(0);
-        encodeInteger(out, 0x20, 5, maxHeaderTableSize);
+        encodeInteger(out, 0x20, 5, (int) maxHeaderTableSize);
```

Comment:
@normanmaurer You should probably also add a comment to this line, explaining why it's safe.